sdk/core/azure-core/tests/test_pipeline.py

```diff
@@ -278,35 +278,34 @@ def test_request_xml(self):
 
         assert request.data == b"<?xml version='1.0' encoding='utf-8'?>\n<root />"
 
-    def test_request_url_with_params(self):
+    def test_request_url_with_query(self):
+
+        request = HttpRequest("GET", url="a/b/c?t=y", query={"g": "h"})
+
+        self.assertIn(request.url, ["a/b/c?g=h&t=y", "a/b/c?t=y&g=h"])
```

Comment:
I'd like to see a test when there was no query in the first place (to also check we add correctly the ?